rpc/src/services/context.rs

```diff
@@ -39,245 +41,219 @@ pub(crate) struct BlockStats {
+            let commit_time: f64 = row.get(3)?;
+
+            Ok((block_id, actions_count, checkout_time, commit_time))
+        }
     )?;
 
-    let mut rows = stmt.query([])?;
-
-    let row = match rows.next()? {
-        Some(row) => row,
-        None => return Ok(None),
-    };
-
-    let stats = ContextStats {
-        actions_count: row.get(0).unwrap_or(0).try_into()?,
-        checkout_context: TimeStats {
-            max: row.get(1).unwrap_or(0.0),
-            mean: row.get(2).unwrap_or(0.0),
-            total_time: row.get(3).unwrap_or(0.0),
-        },
-        commit_context: TimeStats {
-            max: row.get(4).unwrap_or(0.0),
-            mean: row.get(5).unwrap_or(0.0),
-            total_time: row.get(6).unwrap_or(0.0),
-        },
-        operations_context: blocks,
-    };
-
-    return Ok(Some(stats));
-}
-
-fn get_actions_stats(
-    sql: &Connection,
-    mut blocks_map: HashMap<String, BlockStats>,
-) -> Result<Vec<BlockStats>, failure::Error> {
     let mut stmt = sql.prepare(
         "
-         SELECT
-            block_details.action_name,
-            block_details.tezedge_time,
-            blocks.hash AS block_hash
-         FROM
-            block_details
-         JOIN blocks ON block_details.block_id = blocks.id
-            ",
+        SELECT
+          key_root,
+          name,
+          avg(tezedge_time),
+          total(tezedge_time),
+          count(tezedge_time)
+        FROM
+          actions
+        WHERE
+          block_id = ?
+        GROUP BY
+          key_root,
+          name
+        ",
     )?;
 
-    let mut rows = stmt.query([])?;
-
-    while let Some(row) = rows.next()? {
-        let block_hash: String = match row.get(2).unwrap_or(None) {
-            Some(hash) => hash,
-            None => continue,
-        };
-
-        let block = match blocks_map.get_mut(&block_hash) {
-            Some(block) => block,
-            _ => continue,
-        };
+    let mut rows = stmt.query([block_id])?;
 
-        let action_name: String = match row.get(0).unwrap_or(None) {
-            Some(name) => name,
-            None => continue,
-        };
+    let mut map: HashMap<String, Operation> = HashMap::default();
 
-        let value = row.get(1).unwrap_or(0.0);
+    while let Some(row) = rows.next()? {
+        let root: String = row.get(0)?;
+        let action_name: String = row.get(1)?;
+        let _mean: f64 = row.get(2)?;
+        let total: f64 = row.get(3)?;
+        let count: usize = row.get(4)?;
+
+        let entry = map.entry(root.clone()).or_insert_with(|| Operation {
+            data: OperationData {
+                root,
+                mean_time: 0.0,
+                max_time: 0.0,
+                total_time: 0.0,
+                actions_count: 0,
+            },
+            mem: 0.0,
+            find: 0.0,
+            find_tree: 0.0,
+            add: 0.0,
+            add_tree: 0.0,
+            equal: 0.0,
```

Comment:
@sebastiencs  equal/hash/kind/empty/is_empty are not needed (I don't think I even send those)